lib/vsc/mympirun/rm/sched.py

```diff
@@ -42,27 +42,36 @@ def what_sched(requested):
     # The coupler is also a subclass of sched, but we don't want it
     found_sched = [x for x in get_subclasses(Sched) if x.__name__ != 'Coupler']
 
+    # Get local scheduler
+    local_sched = get_local_sched(found_sched)
+
     # first, try to use the scheduler that was requested
     if requested:
         for sched in found_sched:
             if sched._is_sched_for(requested):
                 return sched, found_sched
         LOGGER.warn("%s scheduler was requested, but mympirun failed to find an implementation", requested)
 
+    # if PBS_NODEFILE is not present, use local scheduler
+    if not os.getenv('PBS_NODEFILE'):
+        LOGGER.debug("PBS_NODEFILE not found, using local scheduler")
+        return local_sched, found_sched
```

Comment:
hmm, looking at the logic that is there now, why is this actually needed?

Isn't the loop below supposed to only return a `PBS` scheduler in case we're actually in a PBS job?

It does check for `$PBS_JOBID` rather than `$PBS_NODEFILE` though, but it seems to suggest the actual bug is somewhere else, and we're cheating here?